### PR TITLE
Feature/install with new plugin structure

### PR DESCRIPTION
**Pull request by AlexanderVoss**
Auto detection of the new plugin structure during the install of a plugin.

### CODE REVIEW

File: src/Config.php [hunk 1]
```diff
@@ -53,7 +53,7 @@ public function getRepositories()
         return $this->configArray['repositories'];
     }
 
-    /**
+    /**<
```
Comment (ngoedde) [inline src/Config.php]: Remove this char

File: src/Extensions/Shopware/Plugin/Services/Checkout.php [hunk 2]
```diff
@@ -44,9 +44,9 @@ public function __construct(Utilities $utilities, GitUtil $gitUtil, IoService $i
 
     /**
      * @param Plugin $plugin
-     * @param string $path
-     * @param string $branch
-     * @param bool   $useHttp
+     * @param $path
+     * @param null $branch
```
Comment (ngoedde) [inline src/Extensions/Shopware/Plugin/Services/Checkout.php]: Use correct object-type, null is just the default value.

File: src/Extensions/Shopware/Plugin/Services/Install.php [hunk 3]
```diff
@@ -57,6 +64,26 @@ public function install(Plugin $plugin, $shopwarePath, $inputActivate = false, $
     }
 
     /**
+     * @param $plugin
```
Comment (ngoedde) [inline src/Extensions/Shopware/Plugin/Services/Install.php]: types missing

File: src/Extensions/Shopware/Plugin/Services/Install.php [hunk 4]
```diff
@@ -57,6 +64,26 @@ public function install(Plugin $plugin, $shopwarePath, $inputActivate = false, $
...
+     * @param $shopwarePath
+     */
+    public function checkForNewStructure($plugin, $shopwarePath)
```
Comment (mitelg) [inline src/Extensions/Shopware/Plugin/Services/Install.php]: is there a reason why this method is public?

File: src/Extensions/Shopware/Plugin/Services/Install.php [hunk 5]
```diff
@@ -35,16 +35,23 @@ public function __construct(Checkout $checkout, IoService $ioService)
 
     /**
      * @param Plugin $plugin
-     * @param string $shopwarePath
-     * @param bool   $inputActivate
+     * @param $shopwarePath
```
Comment (mitelg) [inline src/Extensions/Shopware/Plugin/Services/Install.php]: why did you remove the typehint?

File: src/Extensions/Shopware/Plugin/Services/Install.php [hunk 6]
```diff
@@ -57,6 +64,26 @@ public function install(Plugin $plugin, $shopwarePath, $inputActivate = false, $
...
+
+        $destPath = $plugin->module . "/" . $pluginName;
+
+        $absPath = $path . '/' . $destPath;
+
+        if (file_exists($absPath . "/Bootstrap.php")) {
+            return;
+        }
+
+        rename($absPath, $shopwarePath . "/custom/plugins/" . $pluginName);
```
Comment (ngoedde) [inline src/Extensions/Shopware/Plugin/Services/Install.php]: Please use the apostroph instead of quotation marks. There are several occurrences in this PR.

File: src/Extensions/Shopware/Plugin/Services/Checkout.php [hunk 7]
```diff
@@ -44,9 +44,9 @@ public function __construct(Utilities $utilities, GitUtil $gitUtil, IoService $i
 
     /**
      * @param Plugin $plugin
-     * @param string $path
-     * @param string $branch
-     * @param bool   $useHttp
+     * @param $path
```
Comment (mitelg) [inline src/Extensions/Shopware/Plugin/Services/Checkout.php]: typehints?

File: src/Extensions/Shopware/Plugin/Services/Checkout.php [hunk 8]
```diff
@@ -57,6 +57,7 @@ public function checkout(Plugin $plugin, $path, $branch = null, $useHttp = false
         }
         $pluginName = $plugin->name;
         $destPath = $plugin->module . "/" . $plugin->name;
+
```
Comment (ngoedde) [inline src/Extensions/Shopware/Plugin/Services/Checkout.php]: remove this blank line, it does not improve code readability.

File: src/Extensions/Shopware/Plugin/Services/Install.php [hunk 9]
```diff
@@ -60,6 +67,26 @@ public function install(Plugin $plugin, $shopwarePath, $inputActivate = false, $
      * @param Plugin $plugin
      * @param string $shopwarePath
      */
+    private function checkForNewStructure(Plugin $plugin, $shopwarePath)
```
Comment (mitelg) [inline src/Extensions/Shopware/Plugin/Services/Install.php]: please consider the right order of methods: public -> protected -> private

File: src/Extensions/Shopware/Plugin/Services/Checkout.php (hunk 8 above)
Comment (AlexanderVoss) [inline src/Extensions/Shopware/Plugin/Services/Checkout.php]: Done

File: src/Config.php (hunk 1 above)
Comment (AlexanderVoss) [inline src/Config.php]: Done

File: src/Extensions/Shopware/Plugin/Services/Install.php (hunk 5 above)
Comment (AlexanderVoss) [inline src/Extensions/Shopware/Plugin/Services/Install.php]: Done

File: src/Extensions/Shopware/Plugin/Services/Install.php (hunk 4 above)
Comment (AlexanderVoss) [inline src/Extensions/Shopware/Plugin/Services/Install.php]: Done
